Projects/Claims/Sources/Helpers/Network/OdysseyNetworkClient+GetClaims.swift

```diff
@@ -0,0 +1,32 @@
+import Flow
+import Foundation
+import hGraphQL
+
+public protocol GetEntryPointsClaimsClient {
+    func execute() -> Future<[ClaimEntryPointResponseModel]>
+}
+
+extension OdysseyNetworkClient: GetEntryPointsClaimsClient {
+    public func execute() -> Future<[ClaimEntryPointResponseModel]> {
+        return Future { [weak self] completion in
```

Comment:
There are some ways you can avoid the [weak self] here I think.

Here is some pseudo code suggestion:

```swift

Future { completion in
   let bag = DisposeBag()
   
   let task = self.sessionTask.dataTask()
   bag += Disposer { task.cancel() }

  return bag
}
```

This would make sure to cancel the task when the Future gets disposed.